src/languageSupport/languages/flux/lsp/connection.ts

```diff
@@ -18,11 +22,23 @@ import {
 // error reporting
 import {reportErrorThroughHoneyBadger} from 'src/shared/utils/errors'
 
+const ICON_SYNC_ON = 'composition-sync-icon-on'
+const ICON_SYNC_OFF = 'composition-sync-icon-off'
+export const ICON_SYNC_ID = 'schema-composition-sync-icon'
+
+// hardcoded in LSP
+const COMPOSITION_YIELD = '_editor_composition'
+const COMPOSITION_INIT_LINE = 1
+
 class LspConnectionManager {
   private _worker: Worker
+  private _editor: EditorType
   private _model: MonacoTypes.editor.IModel
   private _preludeModel: MonacoTypes.editor.IModel
   private _variables: Variable[] = []
+  private _compositionStyle: string[] = []
+  private _session: SchemaSelection
+  private _callbackSetSession: (schema: SchemaSelection) => void = () => null
```

Comment:
Reasons for state saved. (Feel free to suggest ways to avoid.)
* `_compositionStyle` == required by monaco-editor
   * must provide both previous style, and new style, to the method. (It calc's the diff.)
* `_session`  and `_callbackSetSession`
  * session is our src of truth
  * diff'ing previous vs current
  * callback to set state on session (if need to change anything)